R/datatables.R

```diff
@@ -259,8 +266,21 @@ datatable = function(
   params$caption = captionString(caption)
 
   if (isTRUE(editable)) editable = 'cell'
-  if (is.character(editable)) editable = list(target = editable, disable = list(columns = NULL))
-  if (is.list(editable)) params$editable = editable
+  if (is.character(editable))
+    editable = list(target = editable, disable = list(columns = NULL))
+  if (is.list(editable)) {
+    if (is.null(editable$numeric))
+      editable$numeric =
+        which(unname(vapply(data, is.numeric, logical(1L)))) - 1L
+    else if (identical(editable$numeric, 'none'))
+      editable$numeric = NULL
+    else if (identical(editable$numeric, 'all'))
+      editable$numeric = seq_along(data) - 1L
+    else if (is.null(rn))
+      editable$numeric = editable$numeric - 1
+    editable$numeric = as.list(editable$numeric)
+    params$editable = editable
+  }
```

Comment:
> Not possible from event.button?

I tried a long time to see whether we can catch the origin the event, and I searched a long time with Google and SO. My conclusion was that it is not possible, but I'm ok to check again.

> editable$numeric appeared 10 times in the above code, so I'm wondering if this part could be factored out to a new function to process editable$numeric.

Ok, I will take a look.

Cheers.

Reply:
I've just spent one hour to do other attempts. No way. `event.button` is undefined. I also tried to prevent the change from the `click` event. No way.